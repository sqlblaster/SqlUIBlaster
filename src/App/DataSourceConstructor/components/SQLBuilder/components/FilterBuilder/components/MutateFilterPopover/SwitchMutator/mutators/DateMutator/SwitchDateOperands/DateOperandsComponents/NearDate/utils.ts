import { PluralDateComponent } from './plural-date-components';

export const getThisDateComponentNotation = (
  dateComponent: PluralDateComponent
) => {
  const singularForm = dateComponent.substr(0, dateComponent.length - 1);

  const thisDateNotation =
    singularForm === 'Day' ? 'Today' : `This ${singularForm}`;

  return thisDateNotation.toLowerCase();
};
