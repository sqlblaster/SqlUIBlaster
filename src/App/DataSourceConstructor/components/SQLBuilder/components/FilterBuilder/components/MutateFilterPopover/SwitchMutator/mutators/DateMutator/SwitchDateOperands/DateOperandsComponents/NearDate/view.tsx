import { Checkbox, FormControlLabel, TextField, withStyles } from '@material-ui/core';
import * as React from 'react';
import { CheckedSelect } from '../../../../../CheckedSelect';
import { PluralDateComponent, pluralDateComponents } from './plural-date-components';
import { NearDateViewProps } from './props';
import { nearDateStyles } from './styles';
import { getThisDateComponentNotation } from './utils';

export const NearDate = withStyles(nearDateStyles)((({
  operands: { shiftAmount, dateComponentType, includeCurrentDate },
  operatorView,
  classes,
  renderBottomSection,
  handleShiftAmountChange,
  handleDateComponentTypeChange,
  handleIncludeCurrentDateChange
}) => {
  return (
    <>
      <div className={classes['near-date']}>
        {operatorView}
        <TextField
          className={classes['text-field']}
          margin='dense'
          variant='outlined'
          value={shiftAmount}
          onChange={handleShiftAmountChange}
          inputProps={{ type: 'number', className: classes['shift-amount'] }}
        />
        <CheckedSelect<PluralDateComponent>
          selectedItem={dateComponentType}
          items={pluralDateComponents}
          onSelect={handleDateComponentTypeChange}
        />
      </div>
      {renderBottomSection(
        <FormControlLabel
          className={classes.FormControlLabel}
          control={
            <Checkbox
              id={'include-date-checkbox'}
              checked={includeCurrentDate}
              onChange={handleIncludeCurrentDateChange}
              color='primary'
              style={{ padding: '0px' }}
            />
          }
          htmlFor={'include-date-checkbox'}
          label={
            <div className={classes.label}>
              Include
              <span className={classes['date-component']}>
                {` ${getThisDateComponentNotation(dateComponentType)}`}
              </span>
            </div>
          }
        />
      )}
    </>
  );
}) as React.FC<NearDateViewProps>);
