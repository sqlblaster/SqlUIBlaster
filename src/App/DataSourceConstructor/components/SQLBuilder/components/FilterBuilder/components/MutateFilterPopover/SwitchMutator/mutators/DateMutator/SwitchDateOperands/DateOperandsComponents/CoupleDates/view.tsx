import { withStyles } from '@material-ui/core';
import * as React from 'react';
import { DatePickerByOperandsType } from '../../DatePickerByOperandsType';
import { CoupleDateViewProps } from './props';
import { coupleDateStyles } from './styles';

export const CoupleDate = withStyles(coupleDateStyles)((({
  operatorView,
  operands,
  pickedVariables,
  classes,
  handleVariablePick,
  handleOperandsChange
}) => {
  return (
    <>
      <div className={classes.root}>
        {operatorView}
        <div className={classes.dates}>
          <div className={classes.date} data-testid='between-operand-1'>
            <DatePickerByOperandsType
              operands={operands[0]}
              onOperandsChange={handleOperandsChange(0)}
              variable={pickedVariables[0]}
              onVariablePicked={handleVariablePick(0)}
              limits={
                operands[1].dateType === 'Fixed date'
                  ? {
                      maxDate: operands[1].date,
                      maxDateMessage:
                        'Couldn\'t set time, because selected date is later than second date'
                    }
                  : undefined
              }
            />
          </div>
          <div className={classes.date} data-testid='between-operand-2'>
            <DatePickerByOperandsType
              operands={operands[1]}
              onOperandsChange={handleOperandsChange(1)}
              variable={pickedVariables[1]}
              onVariablePicked={handleVariablePick(1)}
              limits={
                operands[0].dateType === 'Fixed date'
                  ? {
                      minDate: operands[0].date,
                      minDateMessage:
                        'Couldn\'t set time, because selected date is earlier than first date'
                    }
                  : undefined
              }
            />
          </div>
        </div>
      </div>
    </>
  );
}) as React.FC<CoupleDateViewProps>);
