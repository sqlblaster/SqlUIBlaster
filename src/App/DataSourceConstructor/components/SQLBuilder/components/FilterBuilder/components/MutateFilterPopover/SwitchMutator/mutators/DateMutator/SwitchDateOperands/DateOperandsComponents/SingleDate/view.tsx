import { withStyles } from '@material-ui/core';
import * as React from 'react';
import { DatePickerByOperandsType } from '../../DatePickerByOperandsType';
import { SingleDateViewProps } from './props';
import { singleDateStyles } from './styles';

// TODO: strech selects to the full available width
export const SingleDate = withStyles(singleDateStyles)((({
  operatorView,
  operands,
  classes,
  pickedVariables,
  handleVariablePicked,
  handleOperandsChange
}) => (
  <div className={classes.root}>
    {operatorView}
    <DatePickerByOperandsType
      operands={operands}
      onOperandsChange={handleOperandsChange}
      variable={pickedVariables}
      onVariablePicked={handleVariablePicked}
    />
  </div>
)) as React.FC<SingleDateViewProps>);
