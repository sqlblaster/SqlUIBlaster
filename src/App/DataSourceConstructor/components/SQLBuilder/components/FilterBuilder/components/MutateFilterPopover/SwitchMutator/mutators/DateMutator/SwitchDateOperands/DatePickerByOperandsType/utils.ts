import { SingleDateOperands } from '../DateOperandsComponents/SingleDate/model';
import { FixedDateOperands } from './components/FixedDatePicker/model';
import { RelativeDateOperands } from './components/RelativeDatePicker/model';
import { VariableDateOperands } from './components/VariableDatePicker/model';
import { DateType } from './date-operand-types';

export const getNewOperands = (dateType: DateType): SingleDateOperands => {
  switch (dateType) {
    case 'Fixed date':
      return new FixedDateOperands();
    case 'Relative date':
      return new RelativeDateOperands();
    case 'Variable date':
      return new VariableDateOperands();
    default:
      throw new Error('unknown date type is provided');
  }
};
