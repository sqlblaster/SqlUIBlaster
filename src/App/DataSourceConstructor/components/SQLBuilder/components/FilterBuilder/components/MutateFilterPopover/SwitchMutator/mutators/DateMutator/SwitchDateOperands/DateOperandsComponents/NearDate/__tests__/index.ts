import {
  fireEvent,
  getByDisplayValue as getByDisplayValueOn,
  getByText as getByTextOn
} from 'react-testing-library';
import {
  getGeneratedQuery,
  render
} from 'src/App/DataSourceConstructor/components/SQLBuilder/__tests__/render.utils';
import {
  clickAddTheFilter,
  clickUpdateTheFilter
} from 'src/App/DataSourceConstructor/components/SQLBuilder/components/FilterBuilder/__tests__/utils/filter-utils';
import { selectATable } from 'src/App/DataSourceConstructor/components/SQLBuilder/components/TableSelector/__tests__/utils/select-table';
import {
  getByText$,
  regex
} from 'src/App/DataSourceConstructor/test-utils/utils';
import { clickAddAFilter } from '../../../../../../../../AddFilter/__tests__/utils/click-add-a-filter';
import { mutatorPanelId } from '../../../../../../../view';
import { DateOperator } from '../../../../operators';
import {
  PluralDateComponent,
  pluralDateComponents
} from '../plural-date-components';

describe('Previous and Next date operations', () => {
  it(`should render proper sql query on:
        - Both Previous and Next operators
        - All kind of date components
        - Including and excluding the current date component
      should check the shift amount text field,
      should check the proper filter description render
  `, async () => {
    const {
      getByText,
      getByTestId,
      getByLabelText,
      componentNode,
      debug
    } = render();

    const columns = selectATable();
    const dateColumnName = columns[2].name;

    const clickOnFilter = () => {
      fireEvent.click(getByText(regex(dateColumnName)));
    };

    const checkFilterDescription = (
      dateComponent: PluralDateComponent,
      operator: DateOperator = 'Previous'
    ) => {
      expect(
        getByText$(
          componentNode,
          new RegExp(`^${dateColumnName}.*${dateComponent}.*$`, 'i')
        ).textContent
      ).toMatchSnapshot(
        `filter description for ${operator} date operator by ${dateComponent}`
      );
    };

    const checkDateComponent = (
      prevDateComponentIndex: number,
      nextDateComponentIndex: number,
      operator: DateOperator = 'Previous',
      currentDateIncluded = false
    ) => {
      const prevDateComponent = pluralDateComponents[prevDateComponentIndex];
      const nextDateComponent = pluralDateComponents[nextDateComponentIndex];

      clickOnFilter();

      if (nextDateComponent !== prevDateComponent) {
        fireEvent.click(
          getByTextOn(getByTestId(mutatorPanelId), regex(prevDateComponent))
        );
        fireEvent.click(getByText(regex(nextDateComponent)));
      }

      // The current date inclusion text rendered properly
      currentDateIncluded &&
        operator === 'Previous' &&
        expect(getByText(regex('Include')).textContent).toMatchSnapshot(
          `Current date notation for ${operator} date operator by ${nextDateComponent}`
        );

      clickUpdateTheFilter();

      expect(getGeneratedQuery()).toMatchSnapshot(
        `${operator} date operator by ${nextDateComponent} ${
          currentDateIncluded
            ? ` including current ${nextDateComponent.substr(
                0,
                nextDateComponent.length - 1
              )}`
            : ''
        }`
      );
    };

    const checkAllBy = (
      nextOperator: DateOperator,
      prevOperator: DateOperator,
      includeCurrentDate: boolean = false
    ) => {
      if (nextOperator !== prevOperator) {
        clickOnFilter();
        const mutatorPanel = getByTestId(mutatorPanelId);
        fireEvent.click(getByTextOn(mutatorPanel, regex(prevOperator)));
        fireEvent.click(getByTextOn(mutatorPanel, regex(nextOperator)));

        if (includeCurrentDate) {
          enableCurrentDateInclusion();
          changeShiftAmount();
        }

        clickUpdateTheFilter();
      }

      for (let i = 0; i < pluralDateComponents.length; i++) {
        checkDateComponent(
          i - 1 < 0
            ? 2 // Default Days date component
            : i - 1,
          i,
          nextOperator,
          includeCurrentDate
        );
        includeCurrentDate &&
          checkFilterDescription(pluralDateComponents[i], nextOperator);
      }
    };

    const enableCurrentDateInclusion = () => {
      fireEvent.click(getByLabelText(regex('Include')));
    };

    const changeShiftAmount = () => {
      const shiftAmountTextField = getByDisplayValueOn(
        getByTestId(mutatorPanelId),
        '30'
      ) as HTMLInputElement;
      shiftAmountTextField.value = '10';
      fireEvent.input(shiftAmountTextField);
    };

    // Previous should be default one
    clickAddAFilter();
    fireEvent.click(getByText(regex(dateColumnName)));
    clickAddTheFilter();

    expect(getGeneratedQuery()).toMatchSnapshot(
      'Previous operator by Days should be default one'
    );

    // Check all cases for Previous operator
    checkAllBy('Previous', 'Previous');
    // Check all cases for Next operator
    checkAllBy('Next', 'Previous');

    // Check all cases for Previous operator with current date enabled
    checkAllBy('Previous', 'Next', true);
    // Check all cases for Next operator with current date enabled
    checkAllBy('Next', 'Previous', true);
  });
});
