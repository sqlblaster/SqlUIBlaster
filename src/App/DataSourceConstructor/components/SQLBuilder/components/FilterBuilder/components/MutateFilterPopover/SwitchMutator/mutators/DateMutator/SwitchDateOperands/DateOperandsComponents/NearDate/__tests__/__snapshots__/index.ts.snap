// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Previous and Next date operations should render proper sql query on:
        - Both Previous and Next operators
        - All kind of date components
        - Including and excluding the current date component
      should check the shift amount text field,
      should check the proper filter description render
  : Current date notation for Previous date operator by Days 1`] = `"Include today"`;

exports[`Previous and Next date operations should render proper sql query on:
        - Both Previous and Next operators
        - All kind of date components
        - Including and excluding the current date component
      should check the shift amount text field,
      should check the proper filter description render
  : Current date notation for Previous date operator by Hours 1`] = `"Include this hour"`;

exports[`Previous and Next date operations should render proper sql query on:
        - Both Previous and Next operators
        - All kind of date components
        - Including and excluding the current date component
      should check the shift amount text field,
      should check the proper filter description render
  : Current date notation for Previous date operator by Minutes 1`] = `"Include this minute"`;

exports[`Previous and Next date operations should render proper sql query on:
        - Both Previous and Next operators
        - All kind of date components
        - Including and excluding the current date component
      should check the shift amount text field,
      should check the proper filter description render
  : Current date notation for Previous date operator by Months 1`] = `"Include this month"`;

exports[`Previous and Next date operations should render proper sql query on:
        - Both Previous and Next operators
        - All kind of date components
        - Including and excluding the current date component
      should check the shift amount text field,
      should check the proper filter description render
  : Current date notation for Previous date operator by Weeks 1`] = `"Include this week"`;

exports[`Previous and Next date operations should render proper sql query on:
        - Both Previous and Next operators
        - All kind of date components
        - Including and excluding the current date component
      should check the shift amount text field,
      should check the proper filter description render
  : Current date notation for Previous date operator by Years 1`] = `"Include this year"`;

exports[`Previous and Next date operations should render proper sql query on:
        - Both Previous and Next operators
        - All kind of date components
        - Including and excluding the current date component
      should check the shift amount text field,
      should check the proper filter description render
  : Next date operator by Days  1`] = `
"SELECT
  orders1.*
FROM
  \\"orders\\" \\"orders1\\"
WHERE
  date_trunc('Days', (\\"orders1\\".\\"Created At\\")::timestamp) >= date_trunc('Days', (now() + interval '1 Days')::timestamp)
  AND date_trunc('Days', (\\"orders1\\".\\"Created At\\")::timestamp) <= date_trunc('Days', (now() + interval '30 Days')::timestamp)
LIMIT 2000
"
`;

exports[`Previous and Next date operations should render proper sql query on:
        - Both Previous and Next operators
        - All kind of date components
        - Including and excluding the current date component
      should check the shift amount text field,
      should check the proper filter description render
  : Next date operator by Days  including current Day 1`] = `
"SELECT
  orders1.*
FROM
  \\"orders\\" \\"orders1\\"
WHERE
  date_trunc('Days', (\\"orders1\\".\\"Created At\\")::timestamp) >= date_trunc('Days', (now())::timestamp)
  AND date_trunc('Days', (\\"orders1\\".\\"Created At\\")::timestamp) <= date_trunc('Days', (now() + interval '30 Days')::timestamp)
LIMIT 2000
"
`;

exports[`Previous and Next date operations should render proper sql query on:
        - Both Previous and Next operators
        - All kind of date components
        - Including and excluding the current date component
      should check the shift amount text field,
      should check the proper filter description render
  : Next date operator by Hours  1`] = `
"SELECT
  orders1.*
FROM
  \\"orders\\" \\"orders1\\"
WHERE
  date_trunc('Hours', (\\"orders1\\".\\"Created At\\")::timestamp) >= date_trunc('Hours', (now() + interval '1 Hours')::timestamp)
  AND date_trunc('Hours', (\\"orders1\\".\\"Created At\\")::timestamp) <= date_trunc('Hours', (now() + interval '30 Hours')::timestamp)
LIMIT 2000
"
`;

exports[`Previous and Next date operations should render proper sql query on:
        - Both Previous and Next operators
        - All kind of date components
        - Including and excluding the current date component
      should check the shift amount text field,
      should check the proper filter description render
  : Next date operator by Hours  including current Hour 1`] = `
"SELECT
  orders1.*
FROM
  \\"orders\\" \\"orders1\\"
WHERE
  date_trunc('Hours', (\\"orders1\\".\\"Created At\\")::timestamp) >= date_trunc('Hours', (now())::timestamp)
  AND date_trunc('Hours', (\\"orders1\\".\\"Created At\\")::timestamp) <= date_trunc('Hours', (now() + interval '30 Hours')::timestamp)
LIMIT 2000
"
`;

exports[`Previous and Next date operations should render proper sql query on:
        - Both Previous and Next operators
        - All kind of date components
        - Including and excluding the current date component
      should check the shift amount text field,
      should check the proper filter description render
  : Next date operator by Minutes  1`] = `
"SELECT
  orders1.*
FROM
  \\"orders\\" \\"orders1\\"
WHERE
  date_trunc('Minutes', (\\"orders1\\".\\"Created At\\")::timestamp) >= date_trunc('Minutes', (now() + interval '1 Minutes')::timestamp)
  AND date_trunc('Minutes', (\\"orders1\\".\\"Created At\\")::timestamp) <= date_trunc('Minutes', (now() + interval '30 Minutes')::timestamp)
LIMIT 2000
"
`;

exports[`Previous and Next date operations should render proper sql query on:
        - Both Previous and Next operators
        - All kind of date components
        - Including and excluding the current date component
      should check the shift amount text field,
      should check the proper filter description render
  : Next date operator by Minutes  including current Minute 1`] = `
"SELECT
  orders1.*
FROM
  \\"orders\\" \\"orders1\\"
WHERE
  date_trunc('Minutes', (\\"orders1\\".\\"Created At\\")::timestamp) >= date_trunc('Minutes', (now())::timestamp)
  AND date_trunc('Minutes', (\\"orders1\\".\\"Created At\\")::timestamp) <= date_trunc('Minutes', (now() + interval '30 Minutes')::timestamp)
LIMIT 2000
"
`;

exports[`Previous and Next date operations should render proper sql query on:
        - Both Previous and Next operators
        - All kind of date components
        - Including and excluding the current date component
      should check the shift amount text field,
      should check the proper filter description render
  : Next date operator by Months  1`] = `
"SELECT
  orders1.*
FROM
  \\"orders\\" \\"orders1\\"
WHERE
  date_trunc('Months', (\\"orders1\\".\\"Created At\\")::timestamp) >= date_trunc('Months', (now() + interval '1 Months')::timestamp)
  AND date_trunc('Months', (\\"orders1\\".\\"Created At\\")::timestamp) <= date_trunc('Months', (now() + interval '30 Months')::timestamp)
LIMIT 2000
"
`;

exports[`Previous and Next date operations should render proper sql query on:
        - Both Previous and Next operators
        - All kind of date components
        - Including and excluding the current date component
      should check the shift amount text field,
      should check the proper filter description render
  : Next date operator by Months  including current Month 1`] = `
"SELECT
  orders1.*
FROM
  \\"orders\\" \\"orders1\\"
WHERE
  date_trunc('Months', (\\"orders1\\".\\"Created At\\")::timestamp) >= date_trunc('Months', (now())::timestamp)
  AND date_trunc('Months', (\\"orders1\\".\\"Created At\\")::timestamp) <= date_trunc('Months', (now() + interval '30 Months')::timestamp)
LIMIT 2000
"
`;

exports[`Previous and Next date operations should render proper sql query on:
        - Both Previous and Next operators
        - All kind of date components
        - Including and excluding the current date component
      should check the shift amount text field,
      should check the proper filter description render
  : Next date operator by Weeks  1`] = `
"SELECT
  orders1.*
FROM
  \\"orders\\" \\"orders1\\"
WHERE
  date_trunc('Weeks', (\\"orders1\\".\\"Created At\\")::timestamp) >= date_trunc('Weeks', (now() + interval '1 Weeks')::timestamp)
  AND date_trunc('Weeks', (\\"orders1\\".\\"Created At\\")::timestamp) <= date_trunc('Weeks', (now() + interval '30 Weeks')::timestamp)
LIMIT 2000
"
`;

exports[`Previous and Next date operations should render proper sql query on:
        - Both Previous and Next operators
        - All kind of date components
        - Including and excluding the current date component
      should check the shift amount text field,
      should check the proper filter description render
  : Next date operator by Weeks  including current Week 1`] = `
"SELECT
  orders1.*
FROM
  \\"orders\\" \\"orders1\\"
WHERE
  date_trunc('Weeks', (\\"orders1\\".\\"Created At\\")::timestamp) >= date_trunc('Weeks', (now())::timestamp)
  AND date_trunc('Weeks', (\\"orders1\\".\\"Created At\\")::timestamp) <= date_trunc('Weeks', (now() + interval '30 Weeks')::timestamp)
LIMIT 2000
"
`;

exports[`Previous and Next date operations should render proper sql query on:
        - Both Previous and Next operators
        - All kind of date components
        - Including and excluding the current date component
      should check the shift amount text field,
      should check the proper filter description render
  : Next date operator by Years  1`] = `
"SELECT
  orders1.*
FROM
  \\"orders\\" \\"orders1\\"
WHERE
  date_trunc('Years', (\\"orders1\\".\\"Created At\\")::timestamp) >= date_trunc('Years', (now() + interval '1 Years')::timestamp)
  AND date_trunc('Years', (\\"orders1\\".\\"Created At\\")::timestamp) <= date_trunc('Years', (now() + interval '30 Years')::timestamp)
LIMIT 2000
"
`;

exports[`Previous and Next date operations should render proper sql query on:
        - Both Previous and Next operators
        - All kind of date components
        - Including and excluding the current date component
      should check the shift amount text field,
      should check the proper filter description render
  : Next date operator by Years  including current Year 1`] = `
"SELECT
  orders1.*
FROM
  \\"orders\\" \\"orders1\\"
WHERE
  date_trunc('Years', (\\"orders1\\".\\"Created At\\")::timestamp) >= date_trunc('Years', (now())::timestamp)
  AND date_trunc('Years', (\\"orders1\\".\\"Created At\\")::timestamp) <= date_trunc('Years', (now() + interval '30 Years')::timestamp)
LIMIT 2000
"
`;

exports[`Previous and Next date operations should render proper sql query on:
        - Both Previous and Next operators
        - All kind of date components
        - Including and excluding the current date component
      should check the shift amount text field,
      should check the proper filter description render
  : Previous date operator by Days  1`] = `
"SELECT
  orders1.*
FROM
  \\"orders\\" \\"orders1\\"
WHERE
  date_trunc('Days', (\\"orders1\\".\\"Created At\\")::timestamp) >= date_trunc('Days', (now() + interval '-30 Days')::timestamp)
  AND date_trunc('Days', (\\"orders1\\".\\"Created At\\")::timestamp) <= date_trunc('Days', (now() + interval '-1 Days')::timestamp)
LIMIT 2000
"
`;

exports[`Previous and Next date operations should render proper sql query on:
        - Both Previous and Next operators
        - All kind of date components
        - Including and excluding the current date component
      should check the shift amount text field,
      should check the proper filter description render
  : Previous date operator by Days  including current Day 1`] = `
"SELECT
  orders1.*
FROM
  \\"orders\\" \\"orders1\\"
WHERE
  date_trunc('Days', (\\"orders1\\".\\"Created At\\")::timestamp) >= date_trunc('Days', (now() + interval '-30 Days')::timestamp)
  AND date_trunc('Days', (\\"orders1\\".\\"Created At\\")::timestamp) <= date_trunc('Days', (now())::timestamp)
LIMIT 2000
"
`;

exports[`Previous and Next date operations should render proper sql query on:
        - Both Previous and Next operators
        - All kind of date components
        - Including and excluding the current date component
      should check the shift amount text field,
      should check the proper filter description render
  : Previous date operator by Hours  1`] = `
"SELECT
  orders1.*
FROM
  \\"orders\\" \\"orders1\\"
WHERE
  date_trunc('Hours', (\\"orders1\\".\\"Created At\\")::timestamp) >= date_trunc('Hours', (now() + interval '-30 Hours')::timestamp)
  AND date_trunc('Hours', (\\"orders1\\".\\"Created At\\")::timestamp) <= date_trunc('Hours', (now() + interval '-1 Hours')::timestamp)
LIMIT 2000
"
`;

exports[`Previous and Next date operations should render proper sql query on:
        - Both Previous and Next operators
        - All kind of date components
        - Including and excluding the current date component
      should check the shift amount text field,
      should check the proper filter description render
  : Previous date operator by Hours  including current Hour 1`] = `
"SELECT
  orders1.*
FROM
  \\"orders\\" \\"orders1\\"
WHERE
  date_trunc('Hours', (\\"orders1\\".\\"Created At\\")::timestamp) >= date_trunc('Hours', (now() + interval '-30 Hours')::timestamp)
  AND date_trunc('Hours', (\\"orders1\\".\\"Created At\\")::timestamp) <= date_trunc('Hours', (now())::timestamp)
LIMIT 2000
"
`;

exports[`Previous and Next date operations should render proper sql query on:
        - Both Previous and Next operators
        - All kind of date components
        - Including and excluding the current date component
      should check the shift amount text field,
      should check the proper filter description render
  : Previous date operator by Minutes  1`] = `
"SELECT
  orders1.*
FROM
  \\"orders\\" \\"orders1\\"
WHERE
  date_trunc('Minutes', (\\"orders1\\".\\"Created At\\")::timestamp) >= date_trunc('Minutes', (now() + interval '-30 Minutes')::timestamp)
  AND date_trunc('Minutes', (\\"orders1\\".\\"Created At\\")::timestamp) <= date_trunc('Minutes', (now() + interval '-1 Minutes')::timestamp)
LIMIT 2000
"
`;

exports[`Previous and Next date operations should render proper sql query on:
        - Both Previous and Next operators
        - All kind of date components
        - Including and excluding the current date component
      should check the shift amount text field,
      should check the proper filter description render
  : Previous date operator by Minutes  including current Minute 1`] = `
"SELECT
  orders1.*
FROM
  \\"orders\\" \\"orders1\\"
WHERE
  date_trunc('Minutes', (\\"orders1\\".\\"Created At\\")::timestamp) >= date_trunc('Minutes', (now() + interval '-30 Minutes')::timestamp)
  AND date_trunc('Minutes', (\\"orders1\\".\\"Created At\\")::timestamp) <= date_trunc('Minutes', (now())::timestamp)
LIMIT 2000
"
`;

exports[`Previous and Next date operations should render proper sql query on:
        - Both Previous and Next operators
        - All kind of date components
        - Including and excluding the current date component
      should check the shift amount text field,
      should check the proper filter description render
  : Previous date operator by Months  1`] = `
"SELECT
  orders1.*
FROM
  \\"orders\\" \\"orders1\\"
WHERE
  date_trunc('Months', (\\"orders1\\".\\"Created At\\")::timestamp) >= date_trunc('Months', (now() + interval '-30 Months')::timestamp)
  AND date_trunc('Months', (\\"orders1\\".\\"Created At\\")::timestamp) <= date_trunc('Months', (now() + interval '-1 Months')::timestamp)
LIMIT 2000
"
`;

exports[`Previous and Next date operations should render proper sql query on:
        - Both Previous and Next operators
        - All kind of date components
        - Including and excluding the current date component
      should check the shift amount text field,
      should check the proper filter description render
  : Previous date operator by Months  including current Month 1`] = `
"SELECT
  orders1.*
FROM
  \\"orders\\" \\"orders1\\"
WHERE
  date_trunc('Months', (\\"orders1\\".\\"Created At\\")::timestamp) >= date_trunc('Months', (now() + interval '-30 Months')::timestamp)
  AND date_trunc('Months', (\\"orders1\\".\\"Created At\\")::timestamp) <= date_trunc('Months', (now())::timestamp)
LIMIT 2000
"
`;

exports[`Previous and Next date operations should render proper sql query on:
        - Both Previous and Next operators
        - All kind of date components
        - Including and excluding the current date component
      should check the shift amount text field,
      should check the proper filter description render
  : Previous date operator by Weeks  1`] = `
"SELECT
  orders1.*
FROM
  \\"orders\\" \\"orders1\\"
WHERE
  date_trunc('Weeks', (\\"orders1\\".\\"Created At\\")::timestamp) >= date_trunc('Weeks', (now() + interval '-30 Weeks')::timestamp)
  AND date_trunc('Weeks', (\\"orders1\\".\\"Created At\\")::timestamp) <= date_trunc('Weeks', (now() + interval '-1 Weeks')::timestamp)
LIMIT 2000
"
`;

exports[`Previous and Next date operations should render proper sql query on:
        - Both Previous and Next operators
        - All kind of date components
        - Including and excluding the current date component
      should check the shift amount text field,
      should check the proper filter description render
  : Previous date operator by Weeks  including current Week 1`] = `
"SELECT
  orders1.*
FROM
  \\"orders\\" \\"orders1\\"
WHERE
  date_trunc('Weeks', (\\"orders1\\".\\"Created At\\")::timestamp) >= date_trunc('Weeks', (now() + interval '-30 Weeks')::timestamp)
  AND date_trunc('Weeks', (\\"orders1\\".\\"Created At\\")::timestamp) <= date_trunc('Weeks', (now())::timestamp)
LIMIT 2000
"
`;

exports[`Previous and Next date operations should render proper sql query on:
        - Both Previous and Next operators
        - All kind of date components
        - Including and excluding the current date component
      should check the shift amount text field,
      should check the proper filter description render
  : Previous date operator by Years  1`] = `
"SELECT
  orders1.*
FROM
  \\"orders\\" \\"orders1\\"
WHERE
  date_trunc('Years', (\\"orders1\\".\\"Created At\\")::timestamp) >= date_trunc('Years', (now() + interval '-30 Years')::timestamp)
  AND date_trunc('Years', (\\"orders1\\".\\"Created At\\")::timestamp) <= date_trunc('Years', (now() + interval '-1 Years')::timestamp)
LIMIT 2000
"
`;

exports[`Previous and Next date operations should render proper sql query on:
        - Both Previous and Next operators
        - All kind of date components
        - Including and excluding the current date component
      should check the shift amount text field,
      should check the proper filter description render
  : Previous date operator by Years  including current Year 1`] = `
"SELECT
  orders1.*
FROM
  \\"orders\\" \\"orders1\\"
WHERE
  date_trunc('Years', (\\"orders1\\".\\"Created At\\")::timestamp) >= date_trunc('Years', (now() + interval '-30 Years')::timestamp)
  AND date_trunc('Years', (\\"orders1\\".\\"Created At\\")::timestamp) <= date_trunc('Years', (now())::timestamp)
LIMIT 2000
"
`;

exports[`Previous and Next date operations should render proper sql query on:
        - Both Previous and Next operators
        - All kind of date components
        - Including and excluding the current date component
      should check the shift amount text field,
      should check the proper filter description render
  : Previous operator by Days should be default one 1`] = `
"SELECT
  orders1.*
FROM
  \\"orders\\" \\"orders1\\"
WHERE
  date_trunc('Days', (\\"orders1\\".\\"Created At\\")::timestamp) >= date_trunc('Days', (now() + interval '-30 Days')::timestamp)
  AND date_trunc('Days', (\\"orders1\\".\\"Created At\\")::timestamp) <= date_trunc('Days', (now() + interval '-1 Days')::timestamp)
LIMIT 2000
"
`;

exports[`Previous and Next date operations should render proper sql query on:
        - Both Previous and Next operators
        - All kind of date components
        - Including and excluding the current date component
      should check the shift amount text field,
      should check the proper filter description render
  : filter description for Next date operator by Days 1`] = `"Created AtNext 30 Days"`;

exports[`Previous and Next date operations should render proper sql query on:
        - Both Previous and Next operators
        - All kind of date components
        - Including and excluding the current date component
      should check the shift amount text field,
      should check the proper filter description render
  : filter description for Next date operator by Hours 1`] = `"Created AtNext 30 Hours"`;

exports[`Previous and Next date operations should render proper sql query on:
        - Both Previous and Next operators
        - All kind of date components
        - Including and excluding the current date component
      should check the shift amount text field,
      should check the proper filter description render
  : filter description for Next date operator by Minutes 1`] = `"Created AtNext 30 Minutes"`;

exports[`Previous and Next date operations should render proper sql query on:
        - Both Previous and Next operators
        - All kind of date components
        - Including and excluding the current date component
      should check the shift amount text field,
      should check the proper filter description render
  : filter description for Next date operator by Months 1`] = `"Created AtNext 30 Months"`;

exports[`Previous and Next date operations should render proper sql query on:
        - Both Previous and Next operators
        - All kind of date components
        - Including and excluding the current date component
      should check the shift amount text field,
      should check the proper filter description render
  : filter description for Next date operator by Weeks 1`] = `"Created AtNext 30 Weeks"`;

exports[`Previous and Next date operations should render proper sql query on:
        - Both Previous and Next operators
        - All kind of date components
        - Including and excluding the current date component
      should check the shift amount text field,
      should check the proper filter description render
  : filter description for Next date operator by Years 1`] = `"Created AtNext 30 Years"`;

exports[`Previous and Next date operations should render proper sql query on:
        - Both Previous and Next operators
        - All kind of date components
        - Including and excluding the current date component
      should check the shift amount text field,
      should check the proper filter description render
  : filter description for Previous date operator by Days 1`] = `"Created AtPast 30 Days"`;

exports[`Previous and Next date operations should render proper sql query on:
        - Both Previous and Next operators
        - All kind of date components
        - Including and excluding the current date component
      should check the shift amount text field,
      should check the proper filter description render
  : filter description for Previous date operator by Hours 1`] = `"Created AtPast 30 Hours"`;

exports[`Previous and Next date operations should render proper sql query on:
        - Both Previous and Next operators
        - All kind of date components
        - Including and excluding the current date component
      should check the shift amount text field,
      should check the proper filter description render
  : filter description for Previous date operator by Minutes 1`] = `"Created AtPast 30 Minutes"`;

exports[`Previous and Next date operations should render proper sql query on:
        - Both Previous and Next operators
        - All kind of date components
        - Including and excluding the current date component
      should check the shift amount text field,
      should check the proper filter description render
  : filter description for Previous date operator by Months 1`] = `"Created AtPast 30 Months"`;

exports[`Previous and Next date operations should render proper sql query on:
        - Both Previous and Next operators
        - All kind of date components
        - Including and excluding the current date component
      should check the shift amount text field,
      should check the proper filter description render
  : filter description for Previous date operator by Weeks 1`] = `"Created AtPast 30 Weeks"`;

exports[`Previous and Next date operations should render proper sql query on:
        - Both Previous and Next operators
        - All kind of date components
        - Including and excluding the current date component
      should check the shift amount text field,
      should check the proper filter description render
  : filter description for Previous date operator by Years 1`] = `"Created AtPast 30 Years"`;
